cfme/tests/containers/test_containers_data_integrity_properties.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+from cfme.fixtures import pytest_selenium as sel
+from itertools import product
+from cfme.web_ui import CheckboxTable
+from cfme.containers.image import Image
+from cfme.containers.image_registry import ImageRegistry
+from cfme.containers.pod import Pod
+from cfme.containers.provider import Provider
+from cfme.containers.node import Node
+from cfme.containers.container import Container
+from utils import testgen
+from utils.version import current_version
+
+list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table")
+
+pytestmark = [
+    pytest.mark.uncollectif(
+        lambda: current_version() < "5.6"),
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.tier(1)]
+pytest_generate_tests = testgen.generate(
+    testgen.container_providers, scope='function')
+
+CONTAINERS_IMAGES_FIELDS = ['Name', 'Image Id', 'Full Name']
+CONTAINERS_NODES_FIELDS = ['Name', 'Creation timestamp', 'Resource version', 'Number of CPU Cores',
+                           'Memory', 'Max Pods Capacity', 'System BIOS UUID', 'Machine ID',
+                           'Infrastructure Machine ID', 'Runtime version', 'Kubelet version',
+                           'Proxy version', 'Operating System Distribution', 'Kernel version']
+CONTAINERS_FIELDS = ['Name', 'State', 'Restart count', 'Backing Ref (Container ID)', 'Privileged']
+IMAGE_REGISTRY_FIELDS = ['Host']
+
+
+# CMP-9945
+
+@pytest.mark.parametrize('container_fields', product(CONTAINERS_FIELDS))
+def test_containers_data_containers_integrity_properties(container_fields):
+    """ Container details page > Properties fields
+        This test checks correct population of the Properties Fields in Containers' details menu
+        Steps:
+            * Goes to Containers -- > Containers menu
+            * Go through each Container in the menu and check validity of Properties fields
+        """
+    sel.force_navigate('containers_containers')
+    container_name = [r.name.text for r in list_tbl.rows()]
+    container_pod_name = [r.pod_name.text for r in list_tbl.rows()]
+    dictionary = dict(zip(container_name, container_pod_name))
+    for container_name, container_pod_name in dictionary.iteritems():
+        pod = Pod(container_pod_name, Provider)
+        cont = Container(container_name, pod)
+        val = cont.get_detail('Properties', ''.join(container_fields))
+        assert val
+        sel.click("//i[@class='fa fa-arrow-left fa-lg']")
+        sel.click("//a[@data-click='history_choice__history_1']")
```

Comment:
This should be defined somewhere in model and not in tests, because as I said, tests should not define any locators, unless testing a UI glitch for example.